$OpenBSD: patch-ext_openssl_lib_openssl_ssl-internal_rb,v 1.1 2013/07/17 15:51:10 jeremy Exp $

Backport fix for regression introduced by fix for CVE-2013-4073 from
ruby svn r41805.

--- ext/openssl/lib/openssl/ssl-internal.rb.orig	Thu Jun 27 04:13:08 2013
+++ ext/openssl/lib/openssl/ssl-internal.rb	Sun Jul 14 19:24:37 2013
@@ -88,7 +88,7 @@ module OpenSSL
       should_verify_common_name = true
       cert.extensions.each{|ext|
         next if ext.oid != "subjectAltName"
-        id, ostr = OpenSSL::ASN1.decode(ext.to_der).value
+        ostr = OpenSSL::ASN1.decode(ext.to_der).value.last
         sequence = OpenSSL::ASN1.decode(ostr.value)
         sequence.value.each{|san|
           case san.tag
